src/dlgIRC.cpp

```diff
@@ -29,22 +31,49 @@
 #include <QDesktopServices>
 #include <QDir>
 #include <QScrollBar>
+#include <QShortcut>
 #include <QString>
+#include <QTextDocument>
 #include <QTime>
 #include "post_guard.h"
 
 
 dlgIRC::dlgIRC()
 {
+    mInputHistoryMax = 8;
+
     setupUi(this);
-    connection = new IrcConnection(this);
-    irc->setOpenExternalLinks(true);
-    setUnifiedTitleAndToolBarOnMac(true);
-    connect(irc, SIGNAL(anchorClicked(QUrl)), this, SLOT(anchorClicked(QUrl)));
-    connect(connection, SIGNAL(messageReceived(IrcMessage*)), this, SLOT(onMessageReceived(IrcMessage*)));
-    connect(connection, SIGNAL(connected()), this, SLOT(onConnected()));
-    connect(lineEdit, SIGNAL(returnPressed()), this, SLOT(sendMsg()));
+    setupCommandParser();
 
+    connection = new IrcConnection(this);
+    connection->setReconnectDelay(5);
+
+    lineEdit->setAttribute(Qt::WA_MacShowFocusRect, false);
+    ircBrowser->setFocusProxy( lineEdit );
+
+    // nick name completion
+    completer = new IrcCompleter(this);
+    completer->setParser(commandParser);
+    connect(completer, SIGNAL(completed(QString,int)), this, SLOT(slot_nameCompleted(QString,int)));
+    QShortcut* shortcut = new QShortcut(Qt::Key_Tab, this);
+    connect(shortcut, SIGNAL(activated()), this, SLOT(slot_nameCompletion()));
+    QShortcut* shortcut2 = new QShortcut(Qt::Key_Up, this);
+    connect(shortcut2, SIGNAL(activated()), this, SLOT(slot_onHistoryCompletion()));
+
+    connect(lineEdit, SIGNAL(returnPressed()), this, SLOT(slot_onTextEntered()));
+    connect(lineEdit, SIGNAL(textEdited(QString)), this, SLOT(slot_onTextEdited()));
+    connect(ircBrowser, SIGNAL(anchorClicked(QUrl)), this, SLOT(slot_onAnchorClicked(QUrl)));
+    connect(userList, SIGNAL(doubleClicked(QModelIndex)), this, SLOT(slot_onUserActivated(QModelIndex)));
+    connect(connection, SIGNAL(connected()), this, SLOT(slot_onConnected()));
+    connect(connection, SIGNAL(connecting()), this, SLOT(slot_onConnecting()));
+    connect(connection, SIGNAL(disconnected()), this, SLOT(slot_onDisconnected()));
+    connect(connection, SIGNAL(nickNameRequired(QString,QString*)), this, SLOT(slot_nickNameRequired(QString,QString*)));
+    connect(connection, SIGNAL(nickNameChanged(QString)), this, SLOT(slot_nickNameChanged(QString)));
+    connect(connection, SIGNAL(joinMessageReceived(IrcJoinMessage*)), this, SLOT(slot_joinedChannel(IrcJoinMessage*)));
+    connect(connection, SIGNAL(partMessageReceived(IrcPartMessage*)), this, SLOT(slot_partedChannel(IrcPartMessage*)));
+
+    // TODO FIXME : This needs to be updated to use methods provided by Host.
+    qsrand(QTime::currentTime().msec());
```

Comment:
Already covered by https://github.com/Mudlet/Mudlet/blob/development/src/main.cpp#L344 ! No need to do this.